link/src/main/java/com/stripe/android/link/account/CookieStore.kt

```diff
@@ -0,0 +1,42 @@
+package com.stripe.android.link.account
+
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Persistent cookies storage.
+ */
+@Singleton
+internal class CookieStore @Inject constructor(
+    private val store: EncryptedStore
+) {
+    /**
+     * Update authentication session cookie according to the following rules:
+     *
+     * +-----------------------------+---------+
+     * |  cookie                     | Action  |
+     * +-----------------------------+---------+
+     * |  null                       | No-op   |
+     * |-----------------------------|---------|
+     * |  Empty (zero-length) string | Delete  |
+     * |-----------------------------|---------|
+     * |  Any other value            | Store   |
+     * +-----------------------------+---------+
+     */
+    fun updateAuthSessionCookie(cookie: String?) = cookie?.let {
```

Comment:
Does this cookie not need to be associated with a customer?

Reply:
No, this belongs to the account last used in this device. If a different email is used, this will not be valid and a new cookie for the new account will replace the existing one.